### PR TITLE
BetterLimbs v1.1: Hello Despair

### PR DESCRIPTION
**Pull request by Kapu1178**
## About The Pull Request
Rewrites update_icons to be significantly more readable, as well as adds (almost) all the stuff required for nonhumanoid species (grod). While I'm here I redid a good chunk of body temperature code to be more in line with TG, and to fix an issue TG still has. This was for feature parity with Shiptest's version of the framework,

## Why It's Good For The Game
More robust features good.

</details>

## Changelog
:cl:
refactor: Body temperature rewritten
refactor: Human Update_Icons rewritten
add: Hugging transfers body heat
code: Species can now have different temperature threshholds and modifiers.
/:cl:

### CODE REVIEW

File: beestation.dme [hunk 1]
```diff
@@ -2480,7 +2482,6 @@
 #include "code\modules\mob\living\carbon\human\species_types\android.dm"
 #include "code\modules\mob\living\carbon\human\species_types\apid.dm"
 #include "code\modules\mob\living\carbon\human\species_types\corporate.dm"
-#include "code\modules\mob\living\carbon\human\species_types\debug.dm"
```
Comment (DatBoiTim) [inline beestation.dme:2483]: Why?

Comment (DatBoiTim) [inline beestation.dme:2483]: Why?

File: code/__DEFINES/species.dm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#define HUMAN_BODYTEMP_AUTORECOVERY_MINIMUM 12
+///Similar to the HUMAN_BODYTEMP_AUTORECOVERY_DIVISOR, but this is the divisor which is applied at the stage that follows autorecovery. This is the divisor which comes into play when the human's loc temperature is lower than their body temperature. Make it lower to lose bodytemp faster.
+#define HUMAN_BODYTEMP_COLD_DIVISOR 15
+/// Similar to the HUMAN_BODYTEMP_AUTORECOVERY_DIVISOR, but this is the divisor which is applied at the stage that follows autorecovery. This is the divisor which comes into play when the human's loc temperature is higher than their body temperature. Make it lower to gain bodytemp faster.
+#define HUMAN_BODYTEMP_HEAT_DIVISOR 15
+/// The maximum number of degrees that your body can cool in 1 tick, due to the environment, when in a cold area.
+#define HUMAN_BODYTEMP_COOLING_MAX -100
+/// The maximum number of degrees that your body can heat up in 1 tick, due to the environment, when in a hot area.
+#define HUMAN_BODYTEMP_HEATING_MAX 30
+/// The body temperature limit the human body can take before it starts taking damage from heat.
```
Comment (DatBoiTim) [inline code/__DEFINES/species.dm:28]: Unnecessary Refactor Of Define Names

Refactoring the names of vars is really only useful if there's common confusion for the var name. Virology disease vars  vs symptom vars are a prime example of when a rename is warranted.

Comment (DatBoiTim) [inline code/__DEFINES/species.dm:28]: Unecessary Refactor Of Named Defines

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -859,43 +859,55 @@
...
+
+				if((edit_action == "add") && BP)
+					to_chat(usr, "<span class='boldwarning'>[src] already has such bodypart.</span>")
+				else
+					var/limb2add = input(usr, "Select a bodypart type to add", "Add/Replace Bodypart") as null|anything in sortList(limbtypes)
+					var/obj/item/bodypart/new_bp = new limb2add()
+
+					if(new_bp.replace_limb(src, TRUE, TRUE))
+						admin_ticket_log("key_name_admin(usr)] has replaced [src]'s [BP.type] with [new_bp.type]")
+						qdel(BP)
```
Comment (DatBoiTim) [inline code/modules/mob/living/carbon/carbon.dm:906]: Could have easily kept the Switch Statement.

Style wise, should have kept the actions together.

Overall seems like a mess of code.

Comment (DatBoiTim) [inline code/modules/mob/living/carbon/carbon.dm:906]: Could have easily kept the Switch Statement.

Style wise, should have kept the actions together.

File: code/modules/mob/living/carbon/alien/humanoid/life.dm [hunk 4]
```diff
@@ -1,6 +1,6 @@
 
 
-/mob/living/carbon/alien/humanoid/proc/adjust_body_temperature(current, loc_temp, boost)
+/mob/living/carbon/alien/humanoid/proc/adjust_bodytemp(current, loc_temp, boost)
```
Comment (DatBoiTim) [inline code/modules/mob/living/carbon/alien/humanoid/life.dm:3]: Another set of pointless renames clogging the diffs.

Comment (DatBoiTim) [inline code/modules/mob/living/carbon/alien/humanoid/life.dm:3]: Another set of pointless renames clogging the diffs.

File: code/modules/mob/living/carbon/life.dm [hunk 5]
```diff
@@ -585,17 +584,137 @@ GLOBAL_LIST_INIT(ballmer_windows_me_msg, list("Yo man, what if, we like, uh, put
...
+ *
+ * vars:
+ * * amount The amount of degrees to change body temperature by
+ * * min_temp (optional) The minimum body temperature after adjustment
+ * * max_temp (optional) The maximum body temperature after adjustment
+ * * use_insulation (optional) modifies the amount based on the amount of insulation the mob has
+ * * use_steps (optional) Use the body temp divisors and max change rates
+ * * hardsuit_fix (optional) num HUMAN_BODYTEMP_NORMAL - H.bodytemperature Use hardsuit override until hardsuits fix is done...
+ */
+/mob/living/carbon/adjust_bodytemperature(amount, min_temp=0, max_temp=INFINITY, use_insulation=FALSE, use_steps=FALSE, \
```
Comment (DatBoiTim) [inline code/modules/mob/living/carbon/life.dm:701]: Also why is this long form but others aren't?

Having to constantly redefine a proc is silly, plus overhead. Also very inconsistent.

Comment (DatBoiTim) [inline code/modules/mob/living/carbon/life.dm:701]: Also why is this long form but others aren't?

Having to constantly redefine a proc is silly, plus overhead. Also very inconsistent.

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 6]
```diff
@@ -202,42 +197,67 @@
...
-	burn_dam	= round(max(burn_dam - burn, 0), DAMAGE_PRECISION)
-	stamina_dam = round(max(stamina_dam - stamina, 0), DAMAGE_PRECISION)
-	if(owner && updating_health)
-		owner.updatehealth()
-	if(owner.dna && owner.dna.species && (REVIVESBYHEALING in owner.dna.species.species_traits))
-		if(owner.health > 0 && !owner.hellbound)
-			owner.revive(0)
-			owner.cure_husk(0) // If it has REVIVESBYHEALING, it probably can't be cloned. No husk cure.
-	consider_processing()
+	if(brute)
```
Comment (DatBoiTim) [inline code/modules/surgery/bodyparts/_bodyparts.dm:221]: Could easily use a switch()

Comment (DatBoiTim) [inline code/modules/surgery/bodyparts/_bodyparts.dm:221]: Could easily use a switch()

**Review by DatBoiTim — CHANGES_REQUESTED** [review]
Review the comments I had left. A lot of unnecessary diffs created by needless renames that make further review a mess.

File: beestation.dme (hunk 1 above)
Comment (Kapu1178) [reply beestation.dme:2483]: Sprite Sheets was killed, this was used to debug it.

File: code/__DEFINES/species.dm (hunk 2 above)
Comment (Kapu1178) [reply code/__DEFINES/species.dm:28]: Necessary. These are all now species vars.

File: code/modules/mob/living/carbon/alien/humanoid/life.dm (hunk 4 above)
Comment (Kapu1178) [reply code/modules/mob/living/carbon/alien/humanoid/life.dm:3]: Who the fuck cares? Its for consistency with every other use of "bodytemp"

File: code/modules/mob/living/carbon/life.dm (hunk 5 above)
Comment (Kapu1178) [reply code/modules/mob/living/carbon/life.dm:701]: Oversight

File: code/modules/surgery/bodyparts/_bodyparts.dm (hunk 6 above)
Comment (Kapu1178) [reply code/modules/surgery/bodyparts/_bodyparts.dm:221]: No you can't, you can damage more than 1 type at once.

File: code/modules/mob/living/carbon/carbon.dm (hunk 3 above)
Comment (Kapu1178) [reply code/modules/mob/living/carbon/carbon.dm:906]: VV code sucks, I'll clean it up a bit more before I lose patience.

File: code/modules/mob/living/carbon/alien/humanoid/life.dm (hunk 4 above)
Comment (Archanial) [reply code/modules/mob/living/carbon/alien/humanoid/life.dm:3]: if you're renaming procs at least follow basic consistency.
Old name was better in that case

**Pull request closed without merging**